test/wallet-http-test.js

```diff
@@ -269,20 +270,21 @@ describe('Wallet HTTP', function() {
 
     const opens = json.outputs.filter(output => output.covenant.type === types.OPEN);
     assert.equal(opens.length, 1);
+
+    // reset for next test
+    node.mempool.removeListener('tx', handler);
   });
 
   it('should create an open and not broadcast the transaction', async () => {
+    let entered = false;
+    const handler = () => entered = true;
+    node.mempool.on('tx', handler);
+
     const json = await wallet.createOpen({
       name: name,
       broadcast: false
     });
 
-    let entered = false;
-    node.mempool.on('tx', () => {
```

Comment:
By creating the event listener after sending the transaction, it is subject to a race condition where it is possible that the event is never triggered. Good call in moving it before.

Reply:
Yup thanks.